packages/remix-react/routes.tsx

```diff
@@ -258,5 +258,9 @@ function getRedirect(response: Response): Response {
   if (revalidate) {
     headers["X-Remix-Revalidate"] = revalidate;
   }
+  let reloadDocument = response.headers.get("X-Remix-Reload-Document");
+  if (reloadDocument) {
+    headers["X-Remix-Reload-Document"] = reloadDocument;
+  }
```

Comment:
@brophdawg11 do you think it's worth refactoring this into an array + for-loop (or .forEach or .reduce)?

something like:
```ts
let HEADERS_TO_PRESERVE = ["X-Remix-Revalidate", "X-Remix-Reload-Document"];

...

  for (let headerName of HEADERS_TO_PRESERVE) {
    let headerValue = response.headers.get(headerName);
    if (headerValue) {
      headers[headerName] = headerValue;
    }
  }
```

Reply:
nah not yet for the 2nd header - something something kent c dodds avoid hasty abstractions :)